### PR TITLE
Api

### PR DESCRIPTION
**Pull request by simendag**
Implemented various new methods and features for the API, debugged initialization and added some fields to Result model that allows front-end to more easily obtain some name fields. There is a ?depth=bool option for the top level of the API that allows grabbing all the currently available data.

### CODE REVIEW

File: Mandater/Controllers/ElectionController.cs [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        public ElectionController(ElectionContext context, ILogger<ElectionController> logger)
+        {
+            _context = context;
+            _logger = logger;
+        }
+
+        /// <summary>
+        /// Default path method that returns a shallow Country object, showing which countries the API has data on.
+        /// </summary>
+        /// <returns></returns>
```
Comment (Log234) [inline Mandater/Controllers/ElectionController.cs]: Documentation lacks return value.

File: Mandater/Controllers/ElectionController.cs [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+using System.Threading.Tasks;
+using Mandater.Data;
+using Mandater.Models;
+using Microsoft.AspNetCore.Cors;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+
+namespace Mandater.Controllers
```
Comment (Log234) [inline Mandater/Controllers/ElectionController.cs]: Most methods lacks documentations.

File: Mandater/Models/Country.cs [hunk 3]
```diff
@@ -13,11 +13,9 @@ public class Country
         public int CountryId { get; set; }
         public string InternationalName { get; set; }
         [Required]
-        public string ShortName { get; set; }
+        public string CountryCode { get; set; }
 
-        public virtual List<ElectionType> ElectionTypes { get; set; }
-        public virtual List<County> Counties { get; set; }
-        public virtual List<Party> Parties { get; set; }
+        public List<ElectionType> ElectionTypes { get; set; }
```
Comment (Log234) [inline Mandater/Models/Country.cs]: Should be [Required]

File: Mandater/Models/Election.cs [hunk 4]
```diff
@@ -25,12 +25,8 @@ public class Election
...
 
         public int CountryId { get; set; }
-        [Required]
-        public virtual Country Country { get; set; }
 
         public int ElectionTypeId { get; set; }
-        [Required]
-        public virtual ElectionType ElectionType { get; set; }
 
         public virtual List<Result> Results { get; set; }
```
Comment (Log234) [inline Mandater/Models/Election.cs]: Should be [Required]

File: Mandater/Utilities/CSVUtilities.cs [hunk 5]
```diff
@@ -180,18 +180,16 @@ public static Election[] CsvToElectionArray(string filePath, Country country, El
...
                     Year = year,
                     Algorithm = algorithm,
                     FirstDivisor = firstDivisor,
                     Threshold = threshold,
                     Seats = seats,
-                    LevelingSeats = levelingSeats
+                    LevelingSeats = levelingSeats,
+                    Results = new List<Result>()
                 };
+                System.Console.Write("Year" + currentObject.Year);
```
Comment (Log234) [inline Mandater/Utilities/CSVUtilities.cs]: L192: Unnecessary print.

File: Mandater/Controllers/ElectionController.cs [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+                return _context.Countries
+                        .Include(c => c.ElectionTypes)
+                            .ThenInclude(c => c.Elections)
+                                .ThenInclude(c => c.Results);
+            }
+            return _context.Countries;
+        }
+
+        [HttpGet("{countryCode}")]
+        public IEnumerable<ElectionType> GetElectionTypes(string countryCode)
```
Comment (Log234) [inline Mandater/Controllers/ElectionController.cs]: This method crashes when trying to execute [HttpGet("{deep?}")].
Needs to be fixed before merge.

Comment (simendag) [reply Mandater/Controllers/ElectionController.cs]: It doesn't crash for me, are you sure this is the method that is crashing?

**Pull request merged**